src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpUseSearchValues.Fixer.cs

```diff
@@ -0,0 +1,133 @@
+        }
+
+        protected override SyntaxNode ReplaceSearchValuesFieldName(SyntaxNode node)
+        {
+            if (node is FieldDeclarationSyntax fieldDeclaration &&
+                fieldDeclaration.Declaration is { } declaration &&
+                declaration.Variables is [var declarator])
+            {
+                var newDeclarator = declarator.ReplaceToken(declarator.Identifier, declarator.Identifier.WithAdditionalAnnotations(RenameAnnotation.Create()));
+                return fieldDeclaration.WithDeclaration(declaration.WithVariables(new SeparatedSyntaxList<VariableDeclaratorSyntax>().Add(newDeclarator)));
+            }
+
+            return node;
+        }
+
+        protected override SyntaxNode GetDeclaratorInitializer(SyntaxNode syntax)
+        {
+            if (syntax is VariableDeclaratorSyntax variableDeclarator)
+            {
+                return variableDeclarator.Initializer!.Value;
+            }
+
+            if (syntax is PropertyDeclarationSyntax propertyDeclaration)
+            {
+                return CSharpUseSearchValuesAnalyzer.TryGetPropertyGetterExpression(propertyDeclaration)!;
+            }
+
+            throw new InvalidOperationException($"Expected 'VariableDeclaratorSyntax' or 'PropertyDeclarationSyntax', got {syntax.GetType().Name}");
+        }
+
+        // new[] { 'a', 'b', 'c' } => "abc"
+        // new[] { (byte)'a', (byte)'b', (byte)'c' } => "abc"u8
+        // "abc".ToCharArray() => "abc"
+        protected override SyntaxNode? TryReplaceArrayCreationWithInlineLiteralExpression(IOperation operation)
+        {
+            if (operation is IConversionOperation conversion)
+            {
+                operation = conversion.Operand;
+            }
+
+            if (operation is IArrayCreationOperation arrayCreation &&
+                arrayCreation.GetElementType() is { } elementType)
+            {
+                bool isByte = elementType.SpecialType == SpecialType.System_Byte;
+
+                if (isByte &&
+                    (operation.SemanticModel?.Compilation is not CSharpCompilation compilation ||
+                    compilation.LanguageVersion < (LanguageVersion)1100)) // LanguageVersion.CSharp11
+                {
+                    // Can't use Utf8StringLiterals
+                    return null;
+                }
+
+                List<char> values = new();
+
+                if (arrayCreation.Syntax is ExpressionSyntax creationSyntax &&
+                    CSharpUseSearchValuesAnalyzer.IsConstantByteOrCharArrayCreationExpression(operation.SemanticModel!, creationSyntax, values, out _) &&
+                    values.Count <= 128 &&                  // Arbitrary limit to avoid emitting huge literals
+                    !ContainsAnyComments(creationSyntax))   // Avoid removing potentially valuable comments
```

Comment:
This could be confusing as adding/removing comments changes if a code fix is offered or not. Instead of bailing out in presence of comments, consider merging all comments and add it as leading trivia for the generated string literal.

Reply:
Currently the code fixer will always be offered, but may slightly vary in how it'll apply the fix.


E.g. for
```c#
static readonly char[] s_myValues = new[] { 'a', 'b', 'c' };
```
it will replace it with
```c#
static readonly SearchValues<char> s_myValues = SearchValues.Create("abc");
```

but it won't do the `new[] { ... }` => `"abc"` transformation if the original contained a comment. E.g.
```c#
static readonly char[] s_myValues = new[] {
    // We use 'a' here because bla
    'a', 'b', 'c'
};
```
will become
```c#
static readonly SearchValues<char> s_myValues = SearchValues.Create(new[] {
    // We use 'a' here because bla
    'a', 'b', 'c'
});
```